pandas/io/parsers.py

```diff
@@ -370,65 +380,76 @@ def _make_parser_function(name, sep=','):
-                 index_col=None,
                  names=None,
-                 prefix=None,
-                 skiprows=None,
-                 skipfooter=None,
-                 skip_footer=0,
-                 na_values=None,
-                 true_values=None,
-                 false_values=None,
-                 delimiter=None,
-                 converters=None,
-                 dtype=None,
+                 index_col=None,
                  usecols=None,
+                 squeeze=False,
+                 prefix=None,
+                 mangle_dupe_cols=True,
 
+                 # General Parsing Configuration
+                 dtype=None,
                  engine=None,
-                 delim_whitespace=False,
-                 as_recarray=False,
-                 na_filter=True,
-                 compact_ints=False,
-                 use_unsigned=False,
-                 low_memory=_c_parser_defaults['low_memory'],
-                 buffer_lines=None,
-                 warn_bad_lines=True,
-                 error_bad_lines=True,
+                 converters=None,
+                 true_values=None,
+                 false_values=None,
+                 skipinitialspace=False,
+                 skiprows=None,
+                 skipfooter=None,
+                 nrows=None,
 
+                 # NA and Missing Data Handling
+                 na_values=None,
                  keep_default_na=True,
-                 thousands=None,
-                 comment=None,
-                 decimal=b'.',
+                 na_filter=True,
+                 verbose=False,
+                 skip_blank_lines=True,
 
+                 # Datetime Handling
                  parse_dates=False,
+                 infer_datetime_format=False,
                  keep_date_col=False,
-                 dayfirst=False,
                  date_parser=None,
+                 dayfirst=False,
 
-                 memory_map=False,
-                 float_precision=None,
-                 nrows=None,
+                 # Iteration
                  iterator=False,
                  chunksize=None,
 
-                 verbose=False,
+                 # Quoting, Compression, and File Format
+                 compression='infer',
+                 thousands=None,
+                 decimal=b'.',
+                 lineterminator=None,
+                 quotechar='"',
+                 quoting=csv.QUOTE_MINIMAL,
+                 escapechar=None,
+                 comment=None,
                  encoding=None,
-                 squeeze=False,
-                 mangle_dupe_cols=True,
+                 dialect=None,
                  tupleize_cols=False,
-                 infer_datetime_format=False,
-                 skip_blank_lines=True):
+
+                 # Error Handling
+                 error_bad_lines=True,
+                 warn_bad_lines=True,
+
+                 # Deprecated
+                 skip_footer=0,
+
+                 # Internal
```

Comment:
These options don't have corresponding entries in the docstring. For some we may want to add to docstring?